diesel/src/query_source/mod.rs

```diff
@@ -205,3 +208,64 @@ pub trait SizeRestrictedColumn: Column {
     /// Max length of that column
     const MAX_LENGTH: usize;
 }
+
+/// A helper trait to construct new queries based on a struct
+///
+/// ```rust
+/// # include!("../doctest_setup.rs");
+/// #
+/// use schema::users;
+/// use diesel::query_source::QueryType;
+///
+/// // We need to derive `Queryable` + `Identifiable` + `Selectable`
+/// // to enable the `query` method
+/// #[derive(Queryable, PartialEq, Debug, Identifiable, Selectable)]
+/// struct User {
+///     id: i32,
+///     name: String,
+/// }
+///
+/// # fn main() {
+/// #     run_test();
+/// # }
+/// #
+/// # fn run_test() -> QueryResult<()> {
+/// #     use schema::users::dsl::*;
+/// #     let connection = &mut establish_connection();
+/// let all_users = User::query().load(connection)?;
+/// let expected = vec![
+///      User { id: 1, name: "Sean".into() },
+///      User { id: 2, name: "Tess".into() }
+/// ];
+/// assert_eq!(all_users, expected);
+/// // we can join any `QueryDsl` method here:
+/// let sean = User::query().find(1).first(connection)?;
+/// assert_eq!(&sean, &expected[0]);
+/// #     Ok(())
+/// # }
+/// ```
+pub trait QueryType<DB> {
```

Comment:
The name of this trait feels weird to me... Every trait annotates types so `Type` feels like it's only here to bring a distinction with the `Query` trait.

Maybe we could do like `HasTable` and name it `HasQuery`?